analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs

```diff
@@ -37,108 +37,112 @@ public void NullCoalescenceAssignment_NotNull_Noncompliant()
 
         public void NullCoalescenceAssignment_NotNull()
         {
-            string name1 = "";
-            string name2 = null;
-            name1 ??= name2.ToString(); // Noncompliant FIXME (was compliant before) Ok - name1 is not null
-        }
-    }
+            string notNullString = "";
+            string nullString = null;
+            notNullString ??= nullString.ToString(); // Compliant, never visited
 
-    public class NullablePrimitiveType
-    {
-        public void NullablePrimitiveType_FN()
-        {
-            int? i1 = null;
-            i1 = i1 ?? i1.GetHashCode(); // FN - nullable primitive type not supported
+            object notNullObject = new object();
+            object nullObject = null;
+            notNullObject ??= nullObject.ToString(); // Compliant, never visited
         }
-    }
 
-    public class SwitchExpressions
-    {
-        public void Nullable_In_Arm_Noncompliant(string s)
+        public class NullablePrimitiveType
```

Comment:
There's some copy/paste issue with `}`. 
One `}` was deleted after the `NullCoalescenceAssignment_NotNull` that made this `NullablePrimitiveType` a nested class. And one `}` was added at the end of the file where it shouldn't be